### PR TITLE
Added Gateway logging for TF models

### PR DESCRIPTION
**Pull request by S1LV3RJ1NX**

### CODE REVIEW

**Review by chiragjn — APPROVED** [review]

File: models_config.truefoundry.yaml [hunk 1]
```diff
@@ -9,6 +9,12 @@ model_providers:
       embedding_model_ids:
           - "openai-main/text-embedding-ada-002"
       reranking_model_ids: []
+      headers:
+          - "X-TFY-METADATA":
+                {
+                    "tfy_log_request": "true",
+                    "Custom-Metadata": "Cognita-LLM-Request",
+                }
```
Comment (chiragjn) [inline models_config.truefoundry.yaml:17]: Why is this a list?
this should just be `Dict[str, str]`

File: backend/types.py [hunk 2]
```diff
@@ -121,6 +122,18 @@ class ModelProviderConfig(BaseModel):
     reranking_model_ids: List[str]
     api_key_env_var: str
     base_url: Optional[str] = None
+    headers: Optional[List] = None
```
Comment (chiragjn) [inline backend/types.py:125]:
```suggestion
    headers: Dict[str, str] = Field(default_factory=dict)
```

File: backend/types.py [hunk 3]
```diff
@@ -121,6 +122,18 @@ class ModelProviderConfig(BaseModel):
...
+    @property
+    def get_headers(self) -> Dict[str, str]:
+        formatted_headers = {}
+        for header in self.headers:
+            for key, value in header.items():
+                if isinstance(value, dict):
+                    formatted_headers[key] = json.dumps(value)
+                else:
+                    formatted_headers[key] = value
+        return formatted_headers
```
Comment (chiragjn) [inline backend/types.py:136]: No need for this complexity in my opinion, let user's format in the yaml itself

File: models_config.truefoundry.yaml (hunk 1 above)
Comment (S1LV3RJ1NX) [reply models_config.truefoundry.yaml:17]: Like how should the yaml format be?

File: backend/types.py (hunk 3 above)
Comment (S1LV3RJ1NX) [reply backend/types.py:136]: gateway requires dict to be json.dumps()

**Review by chiragjn — APPROVED** [review]

**Pull request merged**